rest/config.go

```diff
@@ -564,65 +567,87 @@ func (config *ServerConfig) setupAndValidateDatabases() error {
 	return nil
 }
 
-func (config *ServerConfig) setupAndValidateLogging(verbose bool, defaultLogFilePath string) error {
+// setupAndValidateLogging sets up and validates logging,
+// and returns a slice of defferred logs to execute later.
+func (config *ServerConfig) setupAndValidateLogging() ([]base.DeferredLog, error) {
 
 	if config.Logging == nil {
 		config.Logging = &base.LoggingConfig{}
 	}
 
 	// populate values from deprecated config options if not set
-	config.deprecatedConfigLoggingFallback(verbose)
+	deferredLogs := config.deprecatedConfigLoggingFallback()
 
 	base.SetRedaction(config.Logging.RedactionLevel)
 
-	if err := config.Logging.Init(defaultLogFilePath); err != nil {
-		return err
+	deferredLogsInit, err := config.Logging.Init(defaultLogFilePath)
+	deferredLogs = append(deferredLogs, deferredLogsInit...)
+	if err != nil {
+		if err == base.ErrUnsetLogFilePath {
+			deferredLogs = append(deferredLogs, func() { base.Errorf(base.KeyAll, "%v", err) })
+		} else {
+			return deferredLogs, err
+		}
 	}
 
 	if config.Logging.DeprecatedDefaultLog == nil {
 		config.Logging.DeprecatedDefaultLog = &base.LogAppenderConfig{}
 	}
 
-	return nil
+	return deferredLogs, nil
 }
 
 // deprecatedConfigLoggingFallback will parse the ServerConfig and try to
 // use older logging config options for backwards compatibility.
-func (config *ServerConfig) deprecatedConfigLoggingFallback(verbose bool) {
+// It will return a slice of deferred log functions to execute at a later time.
+func (config *ServerConfig) deprecatedConfigLoggingFallback() []base.DeferredLog {
+
+	deferredLogs := make([]base.DeferredLog, 0)
```

Comment:
Another style guide issue .. should this be `make(..., 0)` or `deferredLogs := []base.DeferredLog{}`?  I'm fine with either, but I guess it would be good to have a consistent approach.